include/zephyr/drivers/can.h

```diff
@@ -325,6 +325,73 @@ typedef void (*can_state_change_callback_t)(const struct device *dev,
  * For internal driver use only, skip these in public documentation.
  */
 
+/**
+ * @brief Common CAN controller driver configuration.
+ *
+ * This structure is common to all CAN controller drivers and is expected to be the first element in
+ * the object pointed to by the config field in the device structure.
+ */
+struct can_driver_config {
+	/** Pointer to the device structure for the associated CAN transceiver device or NULL. */
+	const struct device *phy;
+	/** The maximum bitrate supported by the CAN controller/transceiver combination. */
+	uint32_t max_bitrate;
+	/** Initial CAN classic/CAN FD arbitration phase bitrate. */
+	uint32_t bus_speed;
+	/** Initial CAN classic/CAN FD arbitration phase sample point in permille. */
+	uint16_t sample_point;
+#ifdef CONFIG_CAN_FD_MODE
+	/** Initial CAN FD data phase bitrate. */
+	uint16_t sample_point_data;
+	/** Initial CAN FD data phase sample point in permille. */
+	uint32_t bus_speed_data;
```

Comment:
Nice catch. Thanks. I switched the fields around in a last minute change to avoid gaps in the structure, but forgot to switch the comments around as well.

Fixed.